src/commands/createFunction/javaSteps/JavaFunctionCreateStep.ts

```diff
@@ -22,36 +21,34 @@ export class JavaFunctionCreateStep extends FunctionCreateStepBase<IFunctionWiza
         return new JavaFunctionCreateStep();
     }
 
-    public async executeCore(context: IFunctionWizardContext & IJavaProjectWizardContext): Promise<string> {
-        const template: IFunctionTemplate = nonNullProp(context, 'functionTemplate');
+    public async executeCore(context: IJavaFunctionWizardContext & IJavaProjectWizardContext): Promise<string> {
+        const template: IJavaFunctionTemplate = nonNullProp(context, 'functionTemplate');
+        const packageName: string = nonNullProp(context, 'javaPackageName');
+        const functionName: string = nonNullProp(context, 'functionName');
 
-        const args: string[] = [];
+        const args: Map<string, string> = new Map<string, string>();
         for (const setting of template.userPromptedSettings) {
             const value = getBindingSetting(context, setting);
             // NOTE: Explicitly checking against undefined. Empty string is a valid value
             if (value !== undefined) {
-                args.push(mavenUtils.formatMavenArg(`D${setting.name}`, value));
+                const fixedValue: string = setting.name === "authLevel" ? String(value).toUpperCase() : String(value);
+                args.set(setting.name, fixedValue);
             }
         }
-
-        const packageName: string = nonNullProp(context, 'javaPackageName');
-        const functionName: string = nonNullProp(context, 'functionName');
-        await mavenUtils.executeMvnCommand(
-            context.telemetry.properties,
-            ext.outputChannel,
-            context.projectPath,
-            'azure-functions:add',
-            '-B',
-            mavenUtils.formatMavenArg('Dfunctions.package', packageName),
-            mavenUtils.formatMavenArg('Dfunctions.name', functionName),
-            mavenUtils.formatMavenArg('Dfunctions.template', removeLanguageFromId(template.id)),
-            ...args
-        );
-
+        args.set("packageName", packageName);
+        args.set("functionName", functionName);
+        args.set("className", functionName.replace('-', '_'));
+        const content: string = substituteParametersInTemplate(template, args);
+        const path: string = getJavaFunctionFilePath(context.projectPath, packageName, functionName);
+        await fse.writeFile(path, content)
         return getJavaFunctionFilePath(context.projectPath, packageName, functionName);
     }
 }
 
-function removeLanguageFromId(id: string): string {
-    return id.split('-')[0];
+function substituteParametersInTemplate(template: IJavaFunctionTemplate, args: Map<string, string>): string {
+    let javaTemplate = template.templateFiles["function.java"];
```

Comment:
Rather than hard-code the file name (which may change) - how about we just use the first file like `Object.values(template.templateFiles)[0]`

Reply:
We may add other files (like `sample.dat`) to java templates, and using the first file may also meet the same issue